### PR TITLE
fix(webservices): improved PAM authentication handling

### PR DESCRIPTION
**Pull request by jeabakker**
So the reported error relates to the authentication issue

### CODE REVIEW

File: mod/web_services/tests/phpunit/integration/Elgg/WebServices/ElggCoreWebServicesApiTest.php [hunk 1]
```diff
@@ -28,9 +28,7 @@ public function testApiAuthenticate() {
 	public function testApiAuthKeyNoKey() {
 		$apikey = new APIKey();
 		
-		$this->expectException(\APIException::class);
-		$this->expectExceptionMessage(elgg_echo('APIException:MissingAPIKey'));
-		$apikey();
+		$this->assertNull($apikey());
 	}
```
Comment (jdalsem) [inline mod/web_services/tests/phpunit/integration/Elgg/WebServices/ElggCoreWebServicesApiTest.php:32]: add test for missing API key scenario (set_input('api_key', null))

**Review by jdalsem — CHANGES_REQUESTED** [review]

**Review by jdalsem — APPROVED** [review]
